engine/docker/docker.go

```diff
@@ -157,10 +157,31 @@ func (e *Docker) Destroy(ctx context.Context, pipelineConfig *spec.PipelineConfi
 	containers := e.containers
 	e.mu.Unlock()
 
+	timeout := time.Second * 30 // similar to k8s default timeout in pods
 	// stop all containers
 	for _, ctrName := range containers {
-		if err := e.client.ContainerKill(ctx, ctrName, "9"); err != nil {
-			logrus.WithField("container", ctrName).WithField("error", err).Warnln("failed to kill container")
+		if err := e.client.ContainerStop(ctx, ctrName, &timeout); err != nil {
+			logrus.WithField("container", ctrName).WithField("error", err).Warnln("failed to stop the container")
+		}
+
+		// Before removing the container we want to be sure that it's in a healthy state to be removed.
+
+		containerStatus, err := e.client.ContainerInspect(ctx, ctrName)
```

Comment:
Curious - would we need this? Could the rest of the code remain as it is? So something like:
Try to do a graceful stop with timeout 30 seconds with `ContainerStop`. After that, the same code that exists today with the `ContainerKill` would execute which would forcefully kill the container.
Another reason for this is that after stopping the containers, the VM is anyways destroyed so we can be more lenient.